tensorflow/python/estimator/canned/tensor_forest.py

```diff
@@ -0,0 +1,209 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Estimator classes for TensorForest."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+
+from tensorflow.python.estimator import estimator
+from tensorflow.python.estimator.canned import head as head_lib
+from tensorflow.python.ops.losses import losses
+from tensorflow.python.summary import summary
+from tensorflow.python.feature_column import feature_column as feature_column_lib
+from tensorflow.python.ops import tensor_forest_ops, math_ops, array_ops
+
+from tensorflow.python.util.tf_export import estimator_export
+
+_ForestHParams = collections.namedtuple('TreeHParams', [
+    'num_output',
+    'n_trees', 'max_nodes', 'num_splits_to_consider',
+    'split_after_samples', 'is_regression',
+])
+
+TREE_PATHS_PREDICTION_KEY = 'tree_paths'
+VARIANCE_PREDICTION_KEY = 'prediction_variance'
+EPSILON = 0.000001
+
+
+def _ensure_logits(head, logits):
```

Comment:
yeah..i can try that,  so first of all it's a forest which is an averaged probability here. if we keep the logit, then.. we are giving the trees with different weight. instead of same weight. so... we can probably try both? not sure which one would be better.